File: projects/project10/Ruby/Bird.rb
# Bird.rb | Defines a Bird superclass to be extended by specific bird
#   sub-classes.
#
# Begun by: Dr. Nelesen, for CS 214 at Calvin College.
# Completed by: Ethan Clark Project10
# Date: April 26, 2016
####################################################

class Bird

	# Define the getter for the name data for the Bird class
  	attr_reader :name

	######################################################
	# initialize(name) creates an instance of class Bird #
	# Recieve: name, a String                            #
	# Output: a Bird type has been initialized with name #
	######################################################
  	def initialize(name)
    		@name = name
  	end

	######################################################
	# call returns the sound that a Bird type makes      #
	# Recieve: NONE                                      #
	# Return: Squaaaaaaaaaaawk!, a String                #
	######################################################
	def call
		"Squaaaaaaawk!"
	end

	#########################################################
	# movement returns the movement that a Bird type does   #
	# Recieve: NONE                                         #
	# Return: moved, a String                               #
	#########################################################
	def movement
		"moved"
	end

	######################################################
	# className returns the name of the Class, Bird      #
	# Recieve: NONE                                      #
	# Return: Bird                                       #
	######################################################
	def className
		self.class.to_s
	end

	####################################################################
	# print returns name, class Name, and the sound a Bird makes       #
	# Recieve: NONE                                                    #
	# Output: a String with name, class Name, movement, and the sound  #
	####################################################################
	def print
		puts name + " " + className + " just " + movement + " and said " + call
	end
end

File: projects/project10/Java/Bird.java
/* Bird.java provides class Bird.
 *
 * Begun by: Dr. Nelesen, CS 214 at Calvin College.
 * Completed by: Ethan Clark Project10
 * Date: April 25, 2016
 */

public class Bird {

    private String myName;

    /*********************************
    * Constructor for Bird Class     *
    * Set name to myName             *
    *********************************/
    public Bird(String name)
    {
	myName = name;
    }
   
    /*********************************
    * Accessor for Bird Class        *
    * Receive: NONE                  *
    * Return: myName                 *
    *********************************/
    public String getName()
    {
	return myName;
    }
   

    /***********************************************
    * call() returns the call of the current Bird  *
    * Receive: NONE                                *
    * Return: "Squaaaaaaaaaaaaaawk!"               *
    ***********************************************/
    public String call()
    {
	return "Squaaaaaaaaaaawk!";
    }

   /*****************************************************************
   * movement() returns the movement type for the Bird class        *
   * Receive: NONE                                                  *
   * Return: "moved"                                                *
   *****************************************************************/
    public String movement()
    {
	return "moved";
    }   

    /********************************************************************
    * print() returns the Bird name and its call                        * 
    * Receive: NONE                                                     *
    * Return: Returns the Bird name, Bird type, movement, and its call  *
    ********************************************************************/
    public void print()
    {
	System.out.println(getName() + ' ' + getClass().getName() +
					" just " + movement() + " and said " + call());
    }
   
}
